### PR TITLE
Update Headline to support displaying bulleted (numbered) text

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-2063

## Summary
Adds a new bulleted text options to the existing Headline component; supersedes the work from https://github.com/boltdesignsystem/bolt/pull/1779;

image

## Details
- [As requested](https://github.com/boltdesignsystem/bolt/pull/1779#pullrequestreview-373756022), the updates to add support to display a round text / number bullet next to the Headline component has been folded in properly.
- The quoted version of Headline was also broken so the work here should fix that as well...
- One thing in particular to note here: the vertical alignment of the text, icons, and bullet itself was particularly tricky to implement so I ended up using CSS custom properties to provide info to the on the parent Headline's line-height (used for setting a minimum height) in case you were wondering where the handful of Core-related style updates came from.
- One _other_ thing to note here: the `.c-bolt-headline__text` and `.c-bolt-headline__text-inner` wrappers are really only needed / used when Icons / Bullets are present. Ideally these would conditionally get output but I'm running up against the clock here... fast follow update perhaps?

## How to test
- Review updated Headline component demos (especially the vertical alignment + Icon behavior) to make sure the updates here visually are as good (if not better) than things were before.

Questions? Comments? Concerns?

### CODE REVIEW

File: docs-site/src/pages/pattern-lab/_patterns/02-components/headline/40-headline-bulleted.twig [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+{% include "@bolt-components-headline/headline.twig" with {
+  size: "xxlarge",
+  text: "bulletSize",
+  tag: "h2",
+} only %}
+
+
+{% include "@bolt-components-headline/headline.twig" with {
+  size: "xlarge",
+  text: "buletSize: 'fixed' (default)",
```
Comment (danielamorse) [inline docs-site/src/pages/pattern-lab/_patterns/02-components/headline/40-headline-bulleted.twig]:
```suggestion
  text: "bulletSize: 'fixed' (default)",
```

File: docs-site/src/pages/pattern-lab/_patterns/02-components/headline/40-headline-bulleted.twig [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+{% include "@bolt-components-headline/eyebrow.twig" with {
+  text: "Bulleted Eyebrow",
+  bulletText: 4,
+} only %}
+
+<hr>
+
+{% include "@bolt-components-headline/headline.twig" with {
+  size: "xlarge",
+  text: "buletSize: 'auto'",
```
Comment (danielamorse) [inline docs-site/src/pages/pattern-lab/_patterns/02-components/headline/40-headline-bulleted.twig]:
```suggestion
  text: "bulletSize: 'auto'",
```

**Comment by sghoweri** [comment]
@danielamorse @mikemai2awesome as mentioned this is ready to re-review -- I push updates some updates earlier per your guys's feedback which should hopefully address everything... 🤞

File: packages/components/bolt-headline/src/_headline.numbered.scss [hunk 3]
```diff
@@ -69,4 +33,94 @@
...
+
+.c-bolt-text--xxsmall,
+.c-bolt-subheadline--xxsmall,
+.c-bolt-headline--xxsmall {
+  .c-bolt-headline__bullet--size-auto {
+    width: $bolt-line-height--xxsmall + em;
+    min-width: $bolt-line-height--xxsmall + em;
+    height: $bolt-line-height--xxsmall + em;
+  }
 }
```
Comment (sghoweri) [inline packages/components/bolt-headline/src/_headline.numbered.scss]: @mikemai2awesome @danielamorse this was literally the approach I took with my Bolt Core style updates via CC custom properties... but without adding an extra 80 lines of CSS that are tied to a fixed # of font sizes in the design system.

We need to ship this however I just wanted to point out that I do not agree with us taking approaches like this (vs something like dynamic vars via CSS custom props) moving forward.

CC @remydenton @adamszalapski

Comment (mikemai2awesome) [reply packages/components/bolt-headline/src/_headline.numbered.scss]: That's a bigger concern with how the headline was built 2 years ago. This is consistent with that without introducing anything new.

**Comment by sghoweri** [comment]
> Approved. Centering looks good now.
> 
> @sghoweri if we had more time to properly test the settings and tools changes, I'd prefer those too. In a time-crunch I'd rather have some repeated code that's scoped to a single component.

@danielamorse Totally agree -- just wanting to point out a good opportunity of something to iterate on

**Comment by sghoweri** [comment]
PR was released with v2.20.0

**Pull request merged**